go/builder.go

```diff
@@ -540,6 +543,24 @@ func (b *Builder) Slot(slotnum int) {
 	b.vtable[slotnum] = UOffsetT(b.Offset())
 }
 
+// FinishWithFileIdentifier finalizes a buffer, pointing to the given `rootTable`.
+// as well as applys a file identifier
+func (b *Builder) FinishWithFileIdentifier(rootTable UOffsetT, fid []byte) {
+	if fid != nil && len(fid) == fileIdentifierLength {
+		// In order to add a file identifier to the flatbuffer message, we need
+		// to prepare an alignment and file identifier length
+		b.Prep(b.minalign, SizeInt32+fileIdentifierLength)
+		// zero out for alignment
+		b.PlaceInt32(0)
```

Comment:
why are these 0-bytes written? these are not written in the other implementations